src/core/MidiInterface.cpp

```diff
@@ -191,12 +179,13 @@ void MidiInterface::releaseNote(int note) {
 	}
 }
 
-void MidiInterface::processMidi(long msg) {
-	int channel = msg & 0xf;
-	int status = (msg >> 4) & 0xf;
-	int data1 = (msg >> 8) & 0xff;
-	int data2 = (msg >> 16) & 0xff;
-	// printf("channel %d status %d data1 %d data2 %d\n", channel, status, data1, data2);
+void MidiInterface::processMidi(std::vector<unsigned char> msg) {
```

Comment:
it is a detail, or maybe I miss something, but using a const reference for the parameter msg can avoid a copy.

Reply:
Who cares, this is preoptimization

Reply:
well, a detail yes, a preoptimization, not sure, just a language detail. But as you say, who cares.

Reply:
Yeah, I need to do this with all my std::strings, but no time for that yet. And it looks uglier  :)